src/components/Search/Search.js

```diff
@@ -7,16 +7,55 @@ const Platform = require("lib/platform");
 const Search = React.createClass({
   getInitialState: function () {
     return {
-      focus: false
+      focus: false,
+      // The following indexes are to keep track of where the user is if they
+      // are navigating the widget with the keyboard.
+      activeIndex: -1,
+      activeSuggestionIndex: -1,
+      activeEngineIndex: -1
     };
   },
   setValueAndSuggestions: function (value) {
+    this.setState({activeIndex: -1, activeSuggestionIndex: -1});
     this.props.dispatch(actions.updateSearchString(value));
     this.props.dispatch(actions.getSuggestions(this.props.Search.currentEngine.name, value));
   },
-  setValueAndClose: function (value) {
-    this.props.dispatch(actions.updateSearchString(value));
-    this.refs.input.blur();
+  getActiveSuggestion: function () {
```

Comment:
I thought for some reason this inherited the parent context like arrow functions, but it doesn't! perhaps we should do a style pass on our components, i think it's ok to keep it consistent for now though